develop/parachain-devs/runtime-development/build-deterministic-runtime.md

```diff
@@ -0,0 +1,130 @@
+
+## Tooling for WebAssembly Runtime
+
+To compile the WebAssembly runtime in a deterministic way, the same tooling that produces the runtime for Polkadot, Kusama, and other Polkadot SDK-based chains can be used. This tooling, referred to collectively as the Substrate runtime toolbox or `srtool`, ensures that the same source code consistently compiles to an identical WebAssembly blob.
+
+The core component of the Substrate Runtime Toolbox (`srtool`) is a Docker container. This container is executed as part of a Docker image. The name of the `srtool` Docker image specifies the version of the Rust compiler used to compile the code included in the image. For example, the image `paritytech/srtool:1.62.0` indicates that the code in the image was compiled with version `1.62.0` of the `rustc` compiler.
+
+## Working with the Docker Container
+
+Because `srtool` is a Docker container, Docker must be available to use it.
+
+The `srtool-cli` package is a command-line utility written in Rust that installs an executable program called `srtool`. This program simplifies the interactions with the `srtool` Docker container.
+
+Over time, the tooling around the `srtool` Docker image has expanded to include the following tools and helper programs:
+
+- [`srtool-cli`](https://github.com/chevdor/srtool-cli){target=\_blank} provides a command-line interface to pull the srtool Docker image, get information about the image and tooling used to interact with it, and build the runtime using the `srtool` Docker container
+- [`subwasm`](https://github.com/chevdor/subwasm){target=\_blank} provides command-line options for working with the metadata and WebAssembly runtime built using srtool. The `subwasm` program is also used internally to perform tasks in the `srtool` image
+- [`srtool-actions`](https://github.com/chevdor/srtool-actions){target=\_blank} provides GitHub actions to integrate builds produced using the `srtool` image with your GitHub CI/CD pipelines
+- [`srtool-app`](https://gitlab.com/chevdor/srtool-app){target=\_blank} provides a simple graphical user interface for building the runtime using the `srtool` Docker image
+
+## Prepare the Environment
+
+To work with the Docker image that executes the code in the `srtool` Docker container, you must have a Docker account and Docker command-line or desktop tools available.
+
+It is recommended to install the `srtool-cli` program to work with the Docker image using a simple command-line interface.
+
+To prepare the environment:
+
+1. Open a terminal shell
+
+2. Verify that that Docker is installed by running the following command:
+
+   ```bash
```

Comment:
Indented text under a list item should have a tab or four spaces. If not, it will break formatting. You have three spaces in most of these kind of sections.